Assets/SequenceSDK/WaaS/Tests/WaaSSessionManagementTests.cs

```diff
@@ -125,5 +125,57 @@ public async Task TestGetSessionAuthProof()
                 tcs.TrySetException(new Exception(error));
             });
         }
+
+        [Test]
+        public async Task GetIdTokenTest()
+        {
+            var tcs = new TaskCompletionSource<bool>();
+
+            var login = SequenceLogin.GetInstance();
+
+            login.OnLoginFailed += (error, method, email, methods) =>
+            {
+
+                Assert.Fail("Login failed: " + error);
+                tcs.TrySetResult(false);
+
+            };
+
+            SequenceWallet.OnWalletCreated += async wallet =>
+            {
+                try
+                {
+                    wallet.OnIdTokenRetrieved += (idToken) =>
+                    {
+                        Assert.IsNotNull(idToken);
+                        Assert.IsFalse(string.IsNullOrWhiteSpace(idToken.IdToken));
+                        Debug.Log(idToken.IdToken);
+                        tcs.TrySetResult(true);
+                    };
+
+                    wallet.OnFailedToRetrieveIdToken += (error) =>
+                    {
+                        Assert.Fail("Failed to retrieve Id token : " + error);
+                        tcs.TrySetResult(false);
+
+                    };
+
+                    await wallet.GetIdToken();
+                }
+                catch (Exception e)
+                {
+                    Assert.Fail("Login failed: " + e.Message);
+                    tcs.TrySetResult(false);
+
+                }
+
+                var result = await tcs.Task;
+
+                Assert.IsTrue(result);
```

Comment:
These should go after `await login.ConnectToWaaSAsGuest();`. The idToken assertions are not being hit before the test is passing.